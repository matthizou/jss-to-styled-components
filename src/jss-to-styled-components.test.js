const pluginTester = require('babel-plugin-tester').default
const jssToStyledComponentsPlugin = require('./babel-plugin-jss-to-styled-components')
const { processCss } = require('./processCss')

const findCss = processCss({ filePath: './assets/360-mini.css' }).findCss

// return
pluginTester({
  pluginName: 'jssToStyledComponentsPlugin',
  plugin: jssToStyledComponentsPlugin,
  babelOptions: require('../babel.config.js'),
  snapshot: true,
  tests: [
    {
      code: `
      import injectSheet from 'react-jss'

      const styles = {
        card: {
          borderTopStyle: 'solid',
          borderTopWidth: '1px',
          borderColor: '#d1d1d1',
        },
        container: {
          composes: 'cssClass1 cssClass2',
          marginTop: '8px',
        },
      }

      const HeaderFooterCard = ({ id, className, classes, footer }) => (
        <section className={classes.container}>
          <div className={classes.card} data-testid="broccoli">🥦</div>
          <div className={cn(classes.card, "mb0")}>🍅</div>
        </section>
      )
    `,
      pluginOptions: {
        findCss,
      },
    },
  ],
})
