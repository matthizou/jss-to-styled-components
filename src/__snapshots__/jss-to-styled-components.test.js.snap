// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jssToStyledComponentsPlugin 1. jssToStyledComponentsPlugin: 1. jssToStyledComponentsPlugin 1`] = `

import injectSheet from 'react-jss'

const styles = {
  card: {
    padding: 4,
  },
}

const HeaderFooterCard = ({ classes }) => (
  <div>
    <span className={classes.card}>🍍</span>
    <div className={classes.card}>🥥</div>
    <span className={classes.card}>🍍</span>
    <Text className={classes.card}>🥥</Text>
  </div>
)

      ↓ ↓ ↓ ↓ ↓ ↓

import injectSheet from 'react-jss'
const Card = styled.span\`
  padding: 4px;
\`

const HeaderFooterCard = ({ classes }) => (
  <div>
    <Card>🍍</Card>
    <Card as="div">🥥</Card>
    <Card>🍍</Card>
    <Card as={Text}>🥥</Card>
  </div>
)


`;
